aws-mfa

```diff
@@ -89,34 +85,41 @@ def validate(args, config):
         key_id = config.get(long_term_name, 'aws_access_key_id')
         access_key = config.get(long_term_name, 'aws_secret_access_key')
     except NoSectionError:
-        logger.error(
+        log_error_and_exit(
             "Long term credentials session '[%s]' is missing. "
             "You must add this section to your credentials file "
             "along with your long term 'aws_access_key_id' and "
             "'aws_secret_access_key'" % (long_term_name,))
-        sys.exit()
     except NoOptionError as e:
-        logger.error(e)
-        sys.exit()
+        log_error_and_exit(e)
 
     if not args.device:
         if os.environ.get('MFA_DEVICE'):
             args.device = os.environ.get('MFA_DEVICE')
         elif config.has_option(long_term_name, 'aws_mfa_device'):
             args.device = config.get(long_term_name, 'aws_mfa_device')
         else:
-            sys.exit('You must provide --device or MFA_DEVICE or set '
-                     '"aws_mfa_device" in ".aws/credentials"')
+            log_error_and_exit(
+                'You must provide --device or MFA_DEVICE or set '
+                '"aws_mfa_device" in ".aws/credentials"')
+
+    if not args.assume_role:
+        if os.environ.get('MFA_ASSUME_ROLE'):
+            args.assume_role = os.environ.get('MFA_ASSUME_ROLE')
 
     if not args.duration:
         if os.environ.get('MFA_STS_DURATION'):
             args.duration = int(os.environ.get('MFA_STS_DURATION'))
+    if args.duration:
```

Comment:
This doesn't handle if both `--duration` and `MFA_STS_DURATION` are omitted. If I remember correctly, the basis of this request is to the set the proper default values. We'll let AWS continue to validate the user's input.

So, I propose:

```
if not args.duration:
    if os.environ.get('MFA_STS_DURATION'):
        args.duration = int(os.environ.get('MFA_STS_DURATION'))
    else:
        if args.assume_rule:
            args.duration = 3600
        else:
            args.duration = 43200
```

Then, removing the validation.